### ISSUE TITLE
Unexpected behaviour after deprecation of parse_streams()

### ISSUE THREAD

**Issue by abuchmueller** [ropensci/rtweet#534]
I've been playing around with `jsonlite::stream_in() `since I'm developing a package which depends on `rtweet` but I'm finding the deprecation of `parse_stream()` troublesome. I've run into multiple issues with `jsonlite::stream_in()` when opening a file
**Notice**: The file in question has been streamed using `stream_tweets()` from the latest development version of `rtweet` from GitHub, while `parse_stream()` has been used on `rweet 0.7.0`.

### Expected behavior

Open preview pane in RStudio

### Reproduce the problem

``` r
zx <- jsonlite::stream_in(file("tweetsaboutbiden.json"))
#> opening file input connection.
#>  Found 500 records... Found 1000 records... Found 1500 records... Found 2000 records... Found 2073 records... Imported 2073 records. Simplifying...
#> closing file input connection.
zx <- tibble::as_tibble(zx)
View(zx)
#> Error in View : 'names' attribute [1] must be the same length as the vector [0]
```
<sup>Created on 2021-03-11 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

I've tested this with multiple json files from different rtweet versions but can't find the issue. There might be an issue with RStudio and the json files from the development version of `rtweet`.

### Expected behavior

Number of retweets in the data should be equal independently of reading method 

``` r
# Parse collected data the "rtweet 1.0 way"
zx <- jsonlite::stream_in(file("tweetsaboutbiden.json")) # rtweet 1.0.0
#> opening file input connection.
#>  Found 500 records... Found 1000 records... Found 1500 records... Found 2000 records... Found 2073 records... Imported 2073 records. Simplifying...
#> closing file input connection.
zx <- tibble::as_tibble(zx)
head(zx, 5)
#> # A tibble: 5 x 36
#>   created_at      id id_str text  display_text_ra… source truncated
#>   <chr>        <dbl> <chr>  <chr> <list>           <chr>  <lgl>    
#> 1 Tue Mar 0… 1.37e18 13693… "@ne… <int [2]>        "<a h… FALSE    
#> 2 Tue Mar 0… 1.37e18 13693… "RT … <NULL>           "<a h… FALSE    
#> 3 Tue Mar 0… 1.37e18 13693… "Joe… <NULL>           "<a h… FALSE    
#> 4 Tue Mar 0… 1.37e18 13693… "RT … <NULL>           "<a h… FALSE    
#> 5 Tue Mar 0… 1.37e18 13693… "RT … <NULL>           "<a h… FALSE    
#> # … with 360 more variables: in_reply_to_status_id <dbl>,
#> #   in_reply_to_status_id_str <chr>, in_reply_to_user_id <dbl>,
#> #   in_reply_to_user_id_str <chr>, in_reply_to_screen_name <chr>,
#> #   user$id <dbl>, $id_str <chr>, $name <chr>, $screen_name <chr>,
#> #   $location <chr>, $url <chr>, $description <chr>, $translator_type <chr>,
#> #   $protected <lgl>, $verified <lgl>, $followers_count <int>,
#> #   $friends_count <int>, $listed_count <int>, $favourites_count <int>,
#> #   $statuses_count <int>, $created_at <chr>, $utc_offset <lgl>,
#> #   $time_zone <lgl>, $geo_enabled <lgl>, $lang <lgl>,
#> #   $contributors_enabled <lgl>, $is_translator <lgl>,
#> #   $profile_background_color <chr>, $profile_background_image_url <chr>,
#> #   $profile_background_image_url_https <chr>, $profile_background_tile <lgl>,
#> #   $profile_link_color <chr>, $profile_sidebar_border_color <chr>,
#> #   $profile_sidebar_fill_color <chr>, $profile_text_color <chr>,
#> #   $profile_use_background_image <lgl>, $profile_image_url <chr>,
#> #   $profile_image_url_https <chr>, $profile_banner_url <chr>,
#> #   $default_profile <lgl>, $default_profile_image <lgl>, $following <lgl>,
#> #   $follow_request_sent <lgl>, $notifications <lgl>, geo <lgl>,
#> #   coordinates <lgl>, place$id <chr>, $url <chr>, $place_type <chr>,
#> #   $name <chr>, $full_name <chr>, $country_code <chr>, $country <chr>,
#> #   $bounding_box$type <chr>, $$coordinates <list>, $attributes <df[,0]>,
#> #   contributors <lgl>, is_quote_status <lgl>, quote_count <int>,
#> #   reply_count <int>, retweet_count <int>, favorite_count <int>,
#> #   entities$hashtags <list>, $urls <list>, $user_mentions <list>,
#> #   $symbols <list>, $media <list>, favorited <lgl>, retweeted <lgl>,
#> #   filter_level <chr>, lang <chr>, timestamp_ms <chr>,
#> #   retweeted_status$created_at <chr>, $id <dbl>, $id_str <chr>, $text <chr>,
#> #   $display_text_range <list>, $source <chr>, $truncated <lgl>,
#> #   $in_reply_to_status_id <dbl>, $in_reply_to_status_id_str <chr>,
#> #   $in_reply_to_user_id <dbl>, $in_reply_to_user_id_str <chr>,
#> #   $in_reply_to_screen_name <chr>, $user$id <dbl>, $$id_str <chr>,
#> #   $$name <chr>, $$screen_name <chr>, $$location <chr>, $$url <chr>,
#> #   $$description <chr>, $$translator_type <chr>, $$protected <lgl>,
#> #   $$verified <lgl>, $$followers_count <int>, $$friends_count <int>,
#> #   $$listed_count <int>, $$favourites_count <int>, $$statuses_count <int>,
#> #   $$created_at <chr>, …
which(zx$retweeted_status$retweeted == FALSE)
#>    [1]    2    4    5    6    8    9   10   11   12   13   15   16   18   22
#>   [15]   23   24   26   27   29   31   32   33   34   35   38   40   41   43
#>   [29]   46   49   50   51   52   53   55   56   58   59   61   62   64   66
#>   [43]   67   69   70   71   72   73   74   77   78   79   80   82   83   84
#>   [57]   85   87   88   89   90   91   92   93   94   95   97   98   99  102
#>   [71]  103  104  105  106  107  108  110  112  113  114  115  116  117  118
#>   [85]  119  120  121  122  123  124  125  126  127  128  131  132  133  134
#>   [99]  135  136  138  140  142  143  144  145  146  147  148  149  150  151
#>  [113]  152  153  154  155  156  159  161  162  163  165  166  167  168  169
#>  [127]  170  171  172  173  174  175  176  178  179  180  181  182  183  185
#>  [141]  186  187  188  190  191  192  193  194  195  196  197  198  199  200
#>  [155]  201  202  203  204  205  206  207  209  210  212  214  216  219  222
#>  [169]  223  226  228  229  230  231  232  233  235  236  238  239  240  243
#>  [183]  244  245  246  248  250  251  253  256  257  258  261  262  263  264
#>  [197]  265  268  270  271  272  273  275  276  279  281  282  283  287  288
#>  [211]  289  290  291  295  296  298  301  302  303  304  305  306  307  309
#>  [225]  311  313  315  317  318  320  322  323  324  327  330  331  332  333
#>  [239]  334  335  336  337  338  339  340  341  342  343  344  347  353  354
#>  [253]  355  356  357  358  359  360  361  362  363  364  366  368  369  370
#>  [267]  372  373  376  377  378  381  384  385  386  387  390  395  396  397
#>  [281]  398  399  400  402  403  405  406  407  408  409  410  411  412  413
#>  [295]  414  415  416  417  418  419  420  421  423  424  425  426  427  428
#>  [309]  431  432  433  434  435  436  437  438  439  441  442  443  445  446
#>  [323]  448  453  454  455  457  458  459  461  462  463  466  467  468  469
#>  [337]  471  472  474  476  477  479  480  481  482  484  486  487  488  489
#>  [351]  490  491  492  493  497  498  499  500  502  505  506  507  508  509
#>  [365]  510  511  512  513  514  515  517  518  520  521  523  524  525  526
#>  [379]  528  529  531  532  533  534  535  536  537  538  539  541  543  544
#>  [393]  545  547  549  550  551  552  553  554  555  556  557  559  560  563
#>  [407]  564  565  566  567  568  570  571  574  575  576  577  580  583  584
#>  [421]  586  587  588  589  590  592  593  595  598  599  601  602  603  604
#>  [435]  605  606  607  610  611  614  615  616  617  618  619  620  622  624
#>  [449]  625  627  628  629  632  633  634  636  637  638  639  640  641  642
#>  [463]  643  646  647  648  649  650  651  653  654  659  660  661  662  666
#>  [477]  668  669  670  672  673  674  677  681  682  684  685  686  687  688
#>  [491]  690  692  693  696  698  699  700  701  702  704  705  706  707  709
#>  [505]  710  711  712  713  714  715  716  718  719  720  722  723  724  726
#>  [519]  727  728  729  733  734  736  737  739  740  741  742  745  746  748
#>  [533]  749  750  751  752  753  755  756  758  759  760  762  763  764  765
#>  [547]  766  767  768  769  770  771  772  773  774  775  777  778  779  780
#>  [561]  781  785  786  788  789  790  791  792  793  794  796  797  798  799
#>  [575]  800  801  802  804  806  807  810  811  812  813  814  817  818  820
#>  [589]  822  823  825  826  827  829  830  832  835  836  837  838  839  841
#>  [603]  842  843  845  846  848  849  850  851  853  855  856  857  858  859
#>  [617]  861  865  866  868  870  871  872  873  874  875  876  877  878  879
#>  [631]  880  881  882  883  885  886  887  888  890  892  893  894  895  896
#>  [645]  897  898  900  901  902  903  904  905  906  907  908  909  911  912
#>  [659]  913  914  915  916  918  919  920  921  922  923  924  925  926  927
#>  [673]  931  933  936  937  938  940  942  943  945  946  947  948  949  951
#>  [687]  952  953  954  955  956  958  959  962  963  965  966  969  970  971
#>  [701]  972  973  975  977  978  979  980  981  983  984  985  986  987  990
#>  [715]  991  992  993  994  995  996  997  998  999 1001 1002 1003 1004 1006
#>  [729] 1007 1008 1009 1010 1013 1014 1015 1016 1018 1019 1020 1021 1023 1025
#>  [743] 1026 1027 1028 1031 1032 1033 1034 1035 1036 1038 1039 1040 1041 1042
#>  [757] 1043 1050 1052 1053 1054 1055 1057 1058 1059 1061 1064 1065 1069 1070
#>  [771] 1071 1072 1073 1074 1075 1076 1079 1080 1081 1082 1083 1084 1086 1087
#>  [785] 1089 1090 1091 1092 1093 1094 1095 1097 1098 1099 1100 1102 1103 1104
#>  [799] 1105 1106 1107 1109 1110 1111 1112 1113 1115 1116 1119 1120 1121 1122
#>  [813] 1123 1124 1126 1127 1128 1129 1130 1132 1134 1135 1136 1139 1140 1141
#>  [827] 1142 1143 1144 1145 1146 1147 1148 1149 1150 1151 1152 1153 1156 1157
#>  [841] 1158 1160 1163 1164 1166 1168 1170 1171 1172 1173 1175 1176 1177 1178
#>  [855] 1179 1180 1181 1184 1185 1186 1187 1188 1189 1190 1191 1192 1193 1194
#>  [869] 1195 1196 1197 1198 1200 1202 1204 1205 1206 1207 1208 1209 1210 1211
#>  [883] 1212 1213 1214 1215 1216 1217 1219 1220 1222 1223 1224 1226 1230 1231
#>  [897] 1232 1234 1235 1236 1237 1238 1239 1242 1243 1246 1248 1249 1251 1252
#>  [911] 1253 1254 1255 1257 1259 1260 1261 1262 1264 1265 1266 1267 1268 1270
#>  [925] 1271 1272 1273 1274 1275 1276 1277 1278 1280 1281 1283 1284 1285 1286
#>  [939] 1287 1288 1289 1290 1291 1293 1294 1295 1297 1298 1300 1303 1304 1305
#>  [953] 1306 1309 1312 1313 1315 1316 1317 1318 1320 1321 1322 1323 1325 1326
#>  [967] 1328 1329 1331 1332 1333 1334 1335 1336 1337 1338 1340 1341 1343 1345
#>  [981] 1347 1348 1349 1350 1351 1353 1355 1356 1357 1359 1361 1362 1364 1365
#>  [995] 1366 1367 1368 1370 1371 1372 1373 1374 1376 1377 1378 1380 1382 1384
#> [1009] 1386 1389 1390 1392 1393 1394 1395 1396 1398 1399 1401 1403 1404 1405
#> [1023] 1406 1408 1409 1411 1412 1414 1415 1416 1417 1419 1420 1422 1423 1424
#> [1037] 1425 1426 1427 1428 1429 1431 1432 1433 1435 1437 1438 1440 1441 1442
#> [1051] 1444 1445 1446 1447 1449 1450 1454 1456 1459 1460 1461 1462 1463 1464
#> [1065] 1465 1467 1469 1470 1471 1472 1473 1474 1475 1476 1477 1478 1479 1480
#> [1079] 1481 1482 1483 1485 1486 1487 1488 1489 1490 1491 1493 1494 1495 1497
#> [1093] 1498 1499 1502 1503 1504 1505 1507 1508 1509 1510 1511 1512 1513 1514
#> [1107] 1515 1517 1518 1520 1521 1522 1524 1525 1526 1527 1528 1529 1531 1532
#> [1121] 1535 1536 1540 1541 1542 1543 1544 1545 1546 1547 1548 1549 1551 1554
#> [1135] 1556 1557 1559 1560 1561 1562 1566 1567 1568 1570 1571 1572 1573 1574
#> [1149] 1575 1576 1577 1578 1579 1580 1581 1583 1585 1587 1588 1590 1591 1592
#> [1163] 1593 1594 1595 1597 1598 1600 1601 1602 1603 1605 1606 1607 1613 1614
#> [1177] 1616 1617 1618 1619 1620 1621 1622 1623 1626 1627 1628 1629 1630 1631
#> [1191] 1632 1633 1634 1635 1636 1640 1641 1642 1643 1644 1645 1646 1647 1648
#> [1205] 1649 1650 1651 1652 1653 1654 1658 1659 1660 1661 1665 1666 1667 1668
#> [1219] 1669 1670 1671 1673 1674 1675 1676 1677 1678 1679 1681 1682 1683 1684
#> [1233] 1686 1687 1688 1695 1696 1697 1698 1699 1701 1703 1704 1705 1708 1709
#> [1247] 1710 1714 1718 1719 1720 1721 1722 1723 1724 1728 1729 1730 1731 1732
#> [1261] 1733 1734 1735 1736 1737 1738 1739 1740 1741 1742 1743 1745 1746 1747
#> [1275] 1748 1749 1750 1753 1754 1756 1757 1758 1761 1764 1765 1767 1768 1771
#> [1289] 1773 1774 1775 1776 1777 1778 1780 1781 1782 1784 1786 1787 1788 1789
#> [1303] 1790 1791 1792 1794 1795 1796 1797 1798 1799 1801 1802 1803 1805 1807
#> [1317] 1811 1812 1813 1814 1816 1818 1821 1822 1824 1825 1826 1827 1829 1832
#> [1331] 1833 1834 1836 1837 1838 1839 1840 1841 1844 1845 1846 1847 1848 1849
#> [1345] 1853 1854 1856 1858 1859 1860 1861 1862 1863 1865 1866 1867 1868 1872
#> [1359] 1873 1874 1876 1877 1878 1879 1880 1881 1883 1885 1886 1887 1888 1890
#> [1373] 1891 1892 1894 1896 1898 1899 1900 1901 1903 1904 1905 1907 1909 1911
#> [1387] 1913 1914 1915 1916 1917 1919 1920 1923 1924 1925 1926 1928 1929 1930
#> [1401] 1931 1932 1934 1935 1936 1937 1938 1939 1941 1942 1943 1944 1945 1947
#> [1415] 1948 1951 1953 1954 1956 1957 1958 1959 1960 1961 1962 1964 1965 1966
#> [1429] 1967 1969 1970 1971 1972 1973 1974 1975 1976 1977 1978 1981 1982 1983
#> [1443] 1984 1985 1986 1987 1988 1989 1990 1991 1992 1993 1994 1998 1999 2000
#> [1457] 2001 2002 2003 2004 2005 2009 2010 2011 2012 2013 2014 2016 2017 2018
#> [1471] 2021 2022 2023 2024 2027 2028 2029 2030 2031 2032 2033 2034 2035 2039
#> [1485] 2040 2041 2042 2044 2045 2047 2048 2049 2051 2052 2054 2055 2056 2060
#> [1499] 2061 2062 2063 2065 2067 2068 2069 2070 2071 2072 2073
```

<sup>Created on 2021-03-11 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

``` r
# the old way
z <- rtweet::parse_stream("tweetsaboutbiden.json")
head(z, 5)
#> # A tibble: 5 x 90
#>   user_id status_id created_at          screen_name text  source
#>   <chr>   <chr>     <dttm>              <chr>       <chr> <chr> 
#> 1 133950… 13693036… 2021-03-09 15:06:36 PakapronSo… "@ne… Twitt…
#> 2 172912… 13693036… 2021-03-09 15:06:36 ddbak69     "Ill… Twitt…
#> 3 172912… 13693043… 2021-03-09 15:09:31 ddbak69     "As … Twitt…
#> 4 129761… 13693036… 2021-03-09 15:06:36 Andreas439… "Joe… Twitt…
#> 5 276222… 13693036… 2021-03-09 15:06:36 PrayagGordy "Pre… Twitt…
#> # … with 84 more variables: display_text_width <dbl>, reply_to_status_id <chr>,
#> #   reply_to_user_id <chr>, reply_to_screen_name <chr>, is_quote <lgl>,
#> #   is_retweet <lgl>, favorite_count <int>, retweet_count <int>,
#> #   quote_count <int>, reply_count <int>, hashtags <list>, symbols <list>,
#> #   urls_url <list>, urls_t.co <list>, urls_expanded_url <list>,
#> #   media_url <list>, media_t.co <list>, media_expanded_url <list>,
#> #   media_type <list>, ext_media_url <list>, ext_media_t.co <list>,
#> #   ext_media_expanded_url <list>, ext_media_type <chr>,
#> #   mentions_user_id <list>, mentions_screen_name <list>, lang <chr>,
#> #   quoted_status_id <chr>, quoted_text <chr>, quoted_created_at <dttm>,
#> #   quoted_source <chr>, quoted_favorite_count <int>,
#> #   quoted_retweet_count <int>, quoted_user_id <chr>, quoted_screen_name <chr>,
#> #   quoted_name <chr>, quoted_followers_count <int>,
#> #   quoted_friends_count <int>, quoted_statuses_count <int>,
#> #   quoted_location <chr>, quoted_description <chr>, quoted_verified <lgl>,
#> #   retweet_status_id <chr>, retweet_text <chr>, retweet_created_at <dttm>,
#> #   retweet_source <chr>, retweet_favorite_count <int>,
#> #   retweet_retweet_count <int>, retweet_user_id <chr>,
#> #   retweet_screen_name <chr>, retweet_name <chr>,
#> #   retweet_followers_count <int>, retweet_friends_count <int>,
#> #   retweet_statuses_count <int>, retweet_location <chr>,
#> #   retweet_description <chr>, retweet_verified <lgl>, place_url <chr>,
#> #   place_name <chr>, place_full_name <chr>, place_type <chr>, country <chr>,
#> #   country_code <chr>, geo_coords <list>, coords_coords <list>,
#> #   bbox_coords <list>, status_url <chr>, name <chr>, location <chr>,
#> #   description <chr>, url <chr>, protected <lgl>, followers_count <int>,
#> #   friends_count <int>, listed_count <int>, statuses_count <int>,
#> #   favourites_count <int>, account_created_at <dttm>, verified <lgl>,
#> #   profile_url <chr>, profile_expanded_url <chr>, account_lang <lgl>,
#> #   profile_banner_url <chr>, profile_background_url <chr>,
#> #   profile_image_url <chr>
which(z$is_retweet == FALSE)
#>   [1]    1    4    8   15   18   20   21   22   23   29   33   35   42   43   45
#>  [16]   49   56   57   58   59   60   61   62   63   64   65   66   68   69   78
#>  [31]   82   89   92   95   98  106  107  111  118  128  133  134  148  149  151
#>  [46]  180  181  189  191  193  210  211  213  217  232  240  246  264  267  271
#>  [61]  273  276  277  278  279  281  282  286  287  289  296  302  303  310  312
#>  [76]  315  317  318  322  323  328  329  330  331  336  337  340  341  343  347
#>  [91]  348  349  350  351  356  357  358  362  363  365  366  376  378  380  382
#> [106]  384  387  389  395  396  399  400  427  428  429  432  433  434  435  436
#> [121]  448  450  456  460  461  462  465  466  467  468  473  474  476  477  478
#> [136]  479  488  492  518  522  523  535  539  542  544  545  546  550  551  554
#> [151]  557  558  564  566  568  571  577  579  588  589  590  596  597  599  600
#> [166]  617  622  625  633  635  646  648  652  656  665  667  668  677  681  682
#> [181]  689  690  692  693  696  701  704  706  707  709  717  718  719  722  723
#> [196]  730  732  738  743  745  746  750  760  761  769  770  771  772  777  778
#> [211]  779  781  785  788  789  790  791  792  796  801  803  806  807  809  813
#> [226]  818  829  833  836  841  842  846  851  852  854  861  862  866  874  877
#> [241]  897  905  906  907  910  918  929  931  935  936  939  943  945  947  950
#> [256]  953  955  956  960  964  967  973  975  981  983  984  985  987  989 1008
#> [271] 1012 1014 1016 1023 1034 1049 1050 1051 1053 1055 1056 1059 1060 1063 1070
#> [286] 1079 1080 1083 1086 1087 1093 1095 1103 1111 1112 1126 1131 1139 1140 1151
#> [301] 1158 1159 1174 1176 1177 1178 1179 1180 1181 1183 1188 1190 1191 1193 1194
#> [316] 1195 1202 1210 1211 1218 1225 1230 1239 1241 1242 1248 1255 1257 1260 1275
#> [331] 1276 1279 1281 1282 1284 1286 1288 1290 1297 1298 1310 1311 1313 1328 1331
#> [346] 1335 1336 1337 1341 1347 1348 1350 1351 1352 1354 1357 1361 1362 1364 1370
#> [361] 1376 1385 1388 1397 1400 1402 1404 1405 1410 1411 1413 1414 1416 1421 1426
#> [376] 1431 1442 1445 1449 1454 1458 1463 1465 1469 1470 1479 1484 1489 1491 1493
#> [391] 1495 1497 1498 1499 1506 1508 1511 1514 1516 1519 1523 1526 1530 1537 1539
#> [406] 1541 1542 1545 1547 1550 1552 1553 1554 1556 1557 1564 1581 1589 1591 1595
#> [421] 1596 1600 1604 1611 1615 1617 1621 1623 1624 1626 1627 1629 1635 1639 1640
#> [436] 1641 1642 1645 1648 1652 1653 1658 1667 1669 1670 1672 1678 1681 1687 1690
#> [451] 1691 1692 1693 1694 1697 1703 1704 1720 1721 1722 1738 1739 1742 1743 1744
#> [466] 1750 1756 1759 1762 1763 1764 1765 1766 1767 1771 1774 1775 1776 1779 1780
#> [481] 1781 1783 1784 1791 1792 1793 1810 1818 1819 1822 1827 1830 1832 1834 1836
#> [496] 1838 1846 1848 1854 1861 1864 1866 1868 1869 1870 1875 1876 1878 1881 1886
#> [511] 1888 1889 1892 1897 1898 1903 1904 1905 1907 1909 1916 1919 1920 1921 1924
#> [526] 1933 1937 1939 1941 1945 1949 1950 1955 1958 1963 1966 1972 1976 1979 1980
#> [541] 1982 1984 1991 1996 2004 2005 2016 2017 2025 2026 2027 2031 2034 2035 2039
#> [556] 2046 2051 2054 2057 2059 2060 2061 2065 2067
```

<sup>Created on 2021-03-11 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### rtweet version

``` r
packageVersion("rtweet")
#> [1] '0.7.0'
```

<sup>Created on 2021-03-11 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### Session info

``` r
sessionInfo()
#> R version 4.0.4 (2021-02-15)
#> Platform: x86_64-apple-darwin17.0 (64-bit)
#> Running under: macOS Big Sur 10.16
#> 
#> Matrix products: default
#> BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
#> LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
#> 
#> locale:
#> [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> loaded via a namespace (and not attached):
#>  [1] compiler_4.0.4    magrittr_1.5      tools_4.0.4       htmltools_0.5.1.1
#>  [5] yaml_2.2.1        stringi_1.4.6     rmarkdown_2.6     highr_0.8        
#>  [9] knitr_1.29        stringr_1.4.0     xfun_0.16         digest_0.6.25    
#> [13] rlang_0.4.10      evaluate_0.14
```

<sup>Created on 2021-03-11 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### Token

``` r
rtweet::get_token()
#> <Token>
#> <oauth_endpoint>
#>  request:   https://api.twitter.com/oauth/request_token
#>  authorize: https://api.twitter.com/oauth/authenticate
#>  access:    https://api.twitter.com/oauth/access_token
#> <oauth_app> rstats2twitter
#>   key:    ****
#>   secret: <hidden>
#> <credentials> oauth_token, oauth_token_secret, user_id, screen_name
#> ---
```

<sup>Created on 2021-03-11 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

So in short, when using `jsonlite::stream_in()` it seems there are 0 retweets in the data (highly implausible) while when using `parse_stream() ` two thirds of the data are retweets.
I've uploaded the [json file in question](https://github.com/ropensci/rtweet/files/6125460/tweetsaboutbiden.json.zip), with which you should be able to reproduce.

**Comment by abuchmueller** [comment]
I've dug deeper into both issues and found out what `parse_stream()` does different than `jsonlite::stream_in()`.

It calls on `tweets_with_users()` which in turn calls `wrangle_retweet_status()` at some point to get the retweet status. Looking into `wrangle_retweet_status()` I found out that it uses

```
  if (has_name_(x, "retweeted_status")) {
    rst <- x[["retweeted_status"]]
  } else {
    rst <- data.frame()
  }
...
x$is_retweet <- !is.na(x$retweet_status_id)
...
```
to detect retweet status.

Now if you use `!is.na(zx$retweeted_status$id_str)` on the data read in by `jsonlite::stream_in()` directly and compare to `zx$is_retweet == TRUE` on the data frame read in by `parse_stream()` you'll get the same number of retweets, which is 1509 in this example.  

### Proof:

``` r
library(rtweet)
zx <- parse_stream("tweetsaboutbiden.json")
sum(zx$is_retweet == TRUE)
#> [1] 1509
```
``` r
library(rtweet)
zx <- jsonlite::stream_in(file("tweetsaboutbiden.json"))
#> opening file input connection.
#>  Found 500 records... Found 1000 records... Found 1500 records... Found 2000 records... Found 2073 records... Imported 2073 records. Simplifying...
#> closing file input connection.
sum(!is.na(zx$retweeted_status$id_str))
#> [1] 1509
```

<sup>Created on 2021-03-17 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

However, `parse_stream()` changes the order of tweets, that's why I was thrown off.

Using ` jsonlite::stream_in()`+  `tweets_with_users()`  does the trick however.

``` r
library(rtweet)
zx <- jsonlite::stream_in(file("tweetsaboutbiden.json"))
#> opening file input connection.
#>  Found 500 records... Found 1000 records... Found 1500 records... Found 2000 records... Found 2073 records... Imported 2073 records. Simplifying...
#> closing file input connection.
zx <- rtweet::tweets_with_users(zx)
sum(zx$is_retweet == TRUE)
#> [1] 1509
```

<sup>Created on 2021-03-17 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

You can see after running `parse_stream()`,  the order of tweets changed slightly so it appears that something is wrong with retweet statuses but in sum they are the same.

``` r
library(rtweet)
zx <- parse_stream("tweetsaboutbiden.json")
zx$text[1:5]
#> [1] "@newsmax A pusher and drama player who supports David Biden's suit and remembers their son being baptized as the son of the royal family."                                                                                                                                                         
#> [2] "Illegal immigrants climbing the incomplete wall at the US/Mexico border and entering the US.\nNo security to be seen now Biden is in charge. https://t.co/DbRGz5XZpH"                                                                                                                              
#> [3] "As recently as last summer, Vanita Gupta called to defund the police. Her allies in the jailbreak industry—who are clamoring for access to the Biden DOJ—are now trying to deny what she said. \n\nBut see for yourself where Vanita Gupta stands on defunding the police: https://t.co/juU9zf08ws"
#> [4] "Joes #NoJokeJoe he is on it"                                                                                                                                                                                                                                                                       
#> [5] "President Trump's first major economic legislation vs. President Biden's, in @TaxPolicyCenter distributional tables: https://t.co/yq8ZNZM6Cf https://t.co/x0LkoenmgF"
```

``` r
library(rtweet)
zx <- jsonlite::stream_in(file("tweetsaboutbiden.json"))
#> opening file input connection.
#>  Found 500 records... Found 1000 records... Found 1500 records... Found 2000 records... Found 2073 records... Imported 2073 records. Simplifying...
#> closing file input connection.
zx$text[1:5]
#> [1] "@newsmax A pusher and drama player who supports David Biden's suit and remembers their son being baptized as the son of the royal family."    
#> [2] "RT @EdwinVcity: Illegal immigrants climbing the incomplete wall at the US/Mexico border and entering the US.\nNo security to be seen now Bid…"
#> [3] "Joes #NoJokeJoe he is on it"                                                                                                                  
#> [4] "RT @jimtankersley: President Trump's first major economic legislation vs. President Biden's, in @TaxPolicyCenter distributional tables: htt…" 
#> [5] "RT @CGTNOfficial: 12 states sue President Biden over climate change order: report https://t.co/CHQNxTNciP"
```
<sup>Created on 2021-03-17 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

So in short while `parse_stream()` works fine with the development version of `rtweet` using `jsonlite::stream_in()` and `tweets_with_users()` also does the trick. This could be added to the documentation or deprecation notice of `parse_stream()` in order to avoid confusion. Also you alternatively could use `jsonlite::stream_in(x, flatten = TRUE)` however this does not produce a nice and tidy tibble like `tweets_with_users()` but an unhandy `n x 363` data frame.

I'm closing the issue now, because the viewer pane issue is more of an RStudio / Twitter issue and the retweets issue, was due to reordering and me incorrectly assuming `x$retweeted_status$retweeted == FALSE` actually returns indices of all retweets, which is confusing but is the way Twitters API returns things. I haven't really found a fix for the preview pane issue but specifying `flatten = TRUE` when using `jsonlite::stream_in()` without `tweets_with_users()`  works as a workaround.

**Comment by llrs** [comment]
I'll reopen the issue to remind us to document it better.

**Comment by hadley** [comment]
Note to self: `stream.Rmd` also needs updating.

**Comment by llrs** [comment]
I'm closing this now that I reviewed the documentation of stream function for #708 (Which will also close https://github.com/ropensci/rtweet/issues/609#issuecomment-884000820) . I undeprecated `parse_stream` and I updated it to return in the same format as other rtweet functions. Once once that PR is merged it should be able possible to do:

```r
zx <- parse_streamtweetsaboutbiden.json")
users_data(zx)
```
